src/GafferOSL/OSLShader.cpp

```diff
@@ -167,7 +167,20 @@ const Gaffer::Plug *OSLShader::correspondingInput( const Gaffer::Plug *output )
 	const StringData *input = IECore::runTimeCast<const StringData>( OSLShader::parameterMetadata( output, "correspondingInput" ) );
 	if( !input )
 	{
-		return NULL;
+		const Gaffer::GraphComponent::ChildContainer &children = output->children();
+		if( children.size() != 1 )
+		{
+			return NULL;
+		}
+
+		if( Gaffer::ConstPlugPtr childPlug = IECore::runTimeCast<Gaffer::Plug>( children[0] ) )
```

Comment:
Better to use a non-owning pointer for this, since we don't need to share ownership - `output->getChild<Plug>( 0 )` should do the job.